pydrive2/files.py

```diff
@@ -344,20 +349,27 @@ def download(fd, request):
                 download(fd, files.get_media(fileId=file_id))
             except errors.HttpError as error:
                 exc = ApiRequestError(error)
-                if (
-                    exc.error["code"] != 403
-                    or exc.GetField("reason") != "fileNotDownloadable"
-                ):
+                code = exc.error["code"]
+                reason = exc.GetField("reason")
+                if code == 403 and reason == "fileNotDownloadable":
+                    mimetype = mimetype or "text/plain"
+                    fd.seek(0)  # just in case `download()` modified `fd`
+                    try:
+                        download(
+                            fd,
+                            files.export_media(
+                                fileId=file_id, mimeType=mimetype
+                            ),
+                        )
+                    except errors.HttpError as error:
+                        raise ApiRequestError(error)
+                elif code == 416 and reason == "requestedRangeNotSatisfiable":
+                    # NOTE: an empty file case, but to fully confirm we would
```

Comment:
I guess it would not be a big deal to fetch metadata just in this case, just to be completely safe. For it means that we'll one extra API call if there is an empty file in cache, right?

Reply:
btw, is it a bug in the MediaDownloader or expected behavior (just curios if figured this out/if we should open a ticket in the google api lib?)

Reply:
Yep, looks like a MediaIoBaseDownload bug, there are a few tickets with unclear conclusions (people say it is fixed, but other's say it is not), but there is a mixup of API versions that people use, so seems like v3 is the one that is working, but the older ones don't. Haven't confirmed v3 though.